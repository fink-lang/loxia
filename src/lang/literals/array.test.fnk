{fink2js} = import '../../testing'
{describe, it, expect, to_match_snapshot} = import '@fink/jest'


describe:: 'array', fn:
  it:: 'compiles', fn:
    expect::
      fink2js`
        array1 = []
        array2 = [1]
        array3 = [1, 2, 4, a(), ...b]
        array4 = [
          a + 1 + 45+
            b + c,
          [1, 2],
          (3 + 3) * 2
        ]
      `
      to_match_snapshot


describe:: 'destructuring array', fn:
  it:: 'compiles', fn:
    expect::
      fink2js`
        [a, b, c] = ni
        [,, d] = ni
        [head, ...tail] = ni
        [...items, last] = '1234'
        [first, second, ...middle, penultimate, end] = '123'
        [ni, ..., nu] = '123'
        [..., nuna] = '123'
      `
      to_match_snapshot



