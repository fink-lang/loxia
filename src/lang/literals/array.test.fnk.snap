// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array compiles 1`] = `
"export const array1 = [];
export const array2 = [1];
export const array3 = [1, 2, 4, a(), a(), ...b];
export const array4 = [a + 1 + 45 + b + c, [1, 2], (3 + 3) * 2];"
`;

exports[`destructuring array compiles 1`] = `
"const [a, b, c] = ni;
const [,, d] = ni;
const [head, ...tail] = ni;

let _do_result;

{
  const [...ˆitems_1] = \`1234\`;
  _do_result = [ˆitems_1.slice(0, -1), ˆitems_1.slice(-1)];
}
const [items, [last]] = _do_result;
_do_result = undefined;

let _do_result2;

{
  const [...ˆitems_2] = \`123\`;
  _do_result2 = [ˆitems_2, ˆitems_2.slice(2, -2), ˆitems_2.slice(-2)];
}
const [[first, second], middle, [penultimate, end]] = _do_result2;
_do_result2 = undefined;

let _do_result3;

{
  const [...ˆitems_3] = \`123\`;
  _do_result3 = [ˆitems_3, ˆitems_3.slice(1, -1), ˆitems_3.slice(-1)];
}
const [[ni],, [nu]] = _do_result3;
_do_result3 = undefined;

let _do_result4;

{
  const [...ˆitems_4] = \`123\`;
  _do_result4 = [ˆitems_4.slice(0, -1), ˆitems_4.slice(-1)];
}
const [, [nuna]] = _do_result4;
_do_result4 = undefined;"
`;
